charts/argo-workflows/templates/controller/workflow-rb.yaml

```diff
@@ -1,19 +1,20 @@
 {{- if .Values.workflow.rbac.create -}}
+  {{- range $namespace := .Values.singeNamespace | ternary (list "") (append .Values.controller.workflowNamespaces .Values.workflow.namespace | compact | uniq) }}
```

Comment:
Actually, looking closer and running some tests, this expression fails in many cases. I came up with this monstrosity instead:
```
  {{- range $namespace := or .Values.singeNamespace false | ternary (list "") (append .Values.controller.workflowNamespaces (coalesce .Values.workflow.namespace .Release.Namespace) | uniq)  }}
```
`ternary` turns out to be very particular about the type of its condition argument and we also want to cover the case when the `.Values.controller.workflowNamespaces` contains `.Values.workflow.namespace`. You'll need to update the expression in all three locations.

Reply:
Fixed in https://github.com/argoproj/argo-helm/pull/824/commits/797fe0a3698a9a27709a3f599425bd6a9f6ef422. That's probably half of the helm keywords collection in an expression! :laughing: